pkg/impersonation/impersonation.go

```diff
@@ -185,6 +193,9 @@ func (i *Impersonator) createNamespace() error {
 	return err
 }
 
+// checkAndUpdateRole checks whether the impersonation clusterrole already exists and whether it has the correct rules.
+// If the role does not exist, the function returns and createRole must be called.
```

Comment:
The smallest of nitpicks - method, rather than function.

Reply:
done